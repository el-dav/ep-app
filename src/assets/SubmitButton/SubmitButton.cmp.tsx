import React, { FC } from 'react';
import { Button, Spin } from 'antd';

import { styled } from 'theme';

const SubmitSpinner = styled(Spin)`
  margin: 3px auto;
  position: absolute;
`;

const Container = styled.div`
  position: relative;
  display: flex;
  align-items: center;
  justify-content: center;
  .ant-btn {
    width: 100px;
    height: 30px;
  }
`;

type Props = {
  className?: string;
  isSubmitting: boolean;
  isDisabled: boolean;
  onSubmit: () => void;
};

const SubmitButton: FC<Props> = ({
  className,
  isSubmitting,
  isDisabled,
  onSubmit
}) => (
  <Container className={`${className || ''}`}>
    <Button
      type="primary"
      disabled={isDisabled || isSubmitting}
      onClick={onSubmit}
    >
      {isSubmitting ? '' : 'Submit'}
    </Button>
    {isSubmitting && <SubmitSpinner />}
  </Container>
);

export default SubmitButton;
