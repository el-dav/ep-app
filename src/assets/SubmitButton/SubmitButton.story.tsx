import React from 'react';
import { storiesOf } from '@storybook/react';
import { boolean } from '@storybook/addon-knobs';
import { action } from '@storybook/addon-actions';

import { ThemeProvider } from 'theme';

import SubmitButton from './SubmitButton.cmp';

storiesOf('SubmitButton', module).add('Default', () => {
  const isSubmitting = boolean('isSubmitting', false);
  const isDisabled = boolean('isDisabled', false);
  const onSubmit = action('onSubmit');

  return (
    <ThemeProvider>
      <SubmitButton
        isSubmitting={isSubmitting}
        isDisabled={isDisabled}
        onSubmit={onSubmit}
      />
    </ThemeProvider>
  );
});
