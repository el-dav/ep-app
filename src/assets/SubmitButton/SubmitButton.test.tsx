import React from 'react';
import renderer from 'react-test-renderer';

import { ThemeProvider } from 'theme';

import SubmitButton from './SubmitButton.cmp';

const onSubmit = () => undefined;

describe('SubmitButton', () => {
  it('renders the Component correctly', () => {
    const tree = renderer
      .create(
        <ThemeProvider>
          <SubmitButton
            onSubmit={onSubmit}
            isDisabled={false}
            isSubmitting={false}
          />
        </ThemeProvider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders the Component correctly while submitting', () => {
    const tree = renderer
      .create(
        <ThemeProvider>
          <SubmitButton onSubmit={onSubmit} isDisabled={false} isSubmitting />
        </ThemeProvider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders the Component correctly while disabled', () => {
    const tree = renderer
      .create(
        <ThemeProvider>
          <SubmitButton onSubmit={onSubmit} isDisabled isSubmitting={false} />
        </ThemeProvider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
