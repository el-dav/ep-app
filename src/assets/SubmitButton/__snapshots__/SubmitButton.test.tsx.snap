// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SubmitButton renders the Component correctly 1`] = `
<div
  className=" css-1yqnfd"
>
  <button
    className="ant-btn ant-btn-primary"
    disabled={false}
    onClick={[Function]}
    type="button"
  >
    <span>
      Submit
    </span>
  </button>
</div>
`;

exports[`SubmitButton renders the Component correctly while disabled 1`] = `
<div
  className=" css-1yqnfd"
>
  <button
    className="ant-btn ant-btn-primary"
    disabled={true}
    onClick={[Function]}
    type="button"
  >
    <span>
      Submit
    </span>
  </button>
</div>
`;

exports[`SubmitButton renders the Component correctly while submitting 1`] = `
<div
  className=" css-1yqnfd"
>
  <button
    className="ant-btn ant-btn-primary"
    disabled={true}
    onClick={[Function]}
    type="button"
  />
  <div
    className="ant-spin ant-spin-spinning css-8mo0vc"
  >
    <span
      className="ant-spin-dot ant-spin-dot-spin"
    >
      <i
        className="ant-spin-dot-item"
      />
      <i
        className="ant-spin-dot-item"
      />
      <i
        className="ant-spin-dot-item"
      />
      <i
        className="ant-spin-dot-item"
      />
    </span>
  </div>
</div>
`;
